src/files-card-body.tsx

```diff
@@ -434,6 +448,63 @@ const getFileType = (file: FolderFileInfo) => {
     }
 };
 
+const FilePermissions = ({ file } : {
+    file: FolderFileInfo,
+}) => {
+    function permissionShortStr(mode: number) {
+        const specialBits = (mode >> 9) & 0b111;
+        const permsStr = [];
+        for (let i = 2; i >= 0; i--) {
+            const offset = i * 3;
+            let shortStr = "";
+            shortStr += (mode & (0b1 << (offset + 2))) ? "r" : "-";
+            shortStr += (mode & (0b1 << (offset + 1))) ? "w" : "-";
+
+            if (mode & (1 << offset)) {
+                if (specialBits & (0b1 << i)) {
+                    shortStr += (i === 0) ? "t" : "s";
+                } else {
+                    shortStr += "x";
+                }
+            } else {
+                if (specialBits & (0b1 << i)) {
+                    shortStr += (i === 0) ? "T" : "S";
+                } else {
+                    shortStr += "-";
+                }
+            }
+
+            permsStr.push(shortStr);
+        }
+
+        return permsStr;
+    }
+
+    const mode = file.mode;
+    if (mode === undefined) {
+        return null;
+    }
+    const permsGroups = [_("Owner"), _("Group"), _("Others")];
+    const tooltip = (
+        <dl className="permissions-tooltip-text">
+            {permsGroups.map((permGroup, i) => {
+                return (
+                    <React.Fragment key={file.name + "-" + permGroup}>
+                        <dt>{permGroup + ":"}</dt>
+                        <dd>{get_permissions(mode >> (6 - 3 * i)).toLowerCase()}</dd>
+                    </React.Fragment>
+                );
+            })}
+        </dl>
+    );
+
+    return (
+        <Tooltip content={tooltip}>
+            <Text component={TextVariants.pre}>{permissionShortStr(mode).join(" ")}</Text>
```

Comment:
Nitpick, `permissionShortStr` this returns an array, I'd join the array when returning there.